charts/presto/values.yaml

```diff
@@ -0,0 +1,172 @@
+  -XX:+ExitOnOutOfMemoryError
+  -Djdk.attach.allowAttachSelf=true
+
+# Common configuration properties, see https://prestodb.io/docs/current/installation/deployment.html#config-properties
+config: |-
+  query.max-memory=1GB
+  query.max-memory-per-node=512MB
+
+# Common log-levels properties, see https://prestodb.io/docs/current/installation/deployment.html#log-levels
+log: |-
+  com.facebook.presto=INFO
+
+coordinator:
+  # Coordinator JVM options (overwrites common JVM options)
+  jvm: ""
+  # Command and arguments to launch coordinator (templated)
+  command: ["sh", "-c", "$PRESTO_HOME/bin/launcher run"]
+  args: ~
+  # Additional coordinator environment variables (templated)
+  env: []
+  envFrom: []
+  # Additional coordinator volumes (templated)
+  volumes: []
+  volumeMounts: []
+  # Additional coordinator extra and init containers (templated)
+  containers: []
+  initContainers: []
+  # Coordinator liveness and readiness probes
+  livenessProbe:
+    initialDelaySeconds: 45
+    periodSeconds: 10
+    timeoutSeconds: 2
+    failureThreshold: 3
+    successThreshold: 1
+  readinessProbe:
+    initialDelaySeconds: 45
+    periodSeconds: 10
+    timeoutSeconds: 2
+    failureThreshold: 3
+    successThreshold: 1
+  # Deployment configuration for coordinator
+  deployment:
+    strategy:
+      type: Recreate
+    revisionHistoryLimit: 3
+  # Coordinator resource limits and requests
+  resources: {}
+  # Constrain coordinator pods to specific nodes (overwrites common constraints)
+  nodeSelector: {}
+  tolerations: []
+  affinity: {}
+  # Coordinator security context (overwrites default security context)
+  securityContext: {}
+
+worker:
+  replicas: 2
+  # Worker JVM options (overwrites common JVM options)
+  jvm: ""
+  # Command and arguments to launch worker (templated)
+  command: ["sh", "-c", "$PRESTO_HOME/bin/launcher run"]
+  args: ~
+  # Additional worker environment variables (templated)
+  env: []
+  envFrom: []
+  # Additional worker volumes (templated)
+  volumes: []
+  volumeMounts: []
+  # Additional worker extra and init containers (templated)
+  containers: []
+  initContainers: []
+  # Worker liveness and readiness probes
+  livenessProbe:
+    initialDelaySeconds: 45
+    periodSeconds: 10
+    timeoutSeconds: 2
+    failureThreshold: 3
+    successThreshold: 1
+  readinessProbe:
+    initialDelaySeconds: 45
+    periodSeconds: 10
+    timeoutSeconds: 2
+    failureThreshold: 3
+    successThreshold: 1
+  # Deployment configuration for worker
+  deployment:
+    strategy:
+      type: Recreate
+    revisionHistoryLimit: 3
+  # Worker resource limits and requests
+  resources: {}
+  # Constrain worker pods to specific nodes (overwrites common constraints)
+  nodeSelector: {}
+  tolerations: []
+  affinity: {}
+  # Worker security context (overwrites default security context)
+  securityContext: {}
+
+catalog:
+  jmx: |-
+    connector.name=jmx
+  tpch: |-
+    connector.name=tpch
+  tpcds: |-
+    connector.name=tpcds
```

Comment:
Sorry for not making one review and jumping through the code - I am not sure if this should be enabled by default, especially if we want to use it for `Prestissimo` as it does not have all of those connectors. I would leave the TPC-H and comment the 2 remaining. You could also add (commented) two bellow ones:
```yaml
  kafka.properties: |-
    connector.name=kafka
    kafka.nodes=kafka:9092
  hive.properties: |-
    connector.name=hive-hadoop2
    hive.metastore.uri=thrift://metastore:9098
```

Reply:
I got suggestions to add these connectors by default:
 - https://github.com/prestodb/presto/pull/18700#issuecomment-1319677465
 - https://github.com/prestodb/presto/pull/18700#issuecomment-1374213428
 - https://github.com/prestodb/presto/pull/18700#issuecomment-1374238990

I would like to not have enabled connectors by default at all in the helm chart, because it is not clear to me why they are needed for all chart users. There is one default connector - `system` and it's enough to test basic functionality for my point of view. For other cases there is the [documentation](https://prestodb.io/docs/current/connector.html) how to configure needed connectors.

Reply:
So it would be the best for this PR to be generic to introduce basic connectors on/off switches so that it can be used/overwritten using cmd line? Making them enabled by default - as you pointed already - is not a good idea.

```yaml
catalog:
  jmx: false
  tpch: false
  tpcds: false
  connectors:
#    kafka.properties: |-
#      connector.name=kafka
#      kafka.nodes=kafka:9092
#    hive.properties: |-
#      connector.name=hive-hadoop2
#      hive.metastore.uri=thrift://metastore:9098
```

and then:

```yaml
data:
{{- if .Values.catalog.jmx }}
  jmx.properties: |-
    connector.name=jmx
{{- end }}
{{- if .Values.catalog.tpch }}
  tpch.properties: |-
    connector.name=tpch
{{- end }}
{{- if .Values.catalog.tpcds}}
  tpcds.properties: |-
    connector.name=tpcds
{{- end }}
  {{- range $name, $properties := .Values.catalog.connectors }}
  {{ $name }}.properties: |-
    {{- $properties | nindent 4 }}
  {{- end }}
```

But this is not at all an elegant resolution of the issue - I personally would vote for not adding any default connectors instead of adding some of them or implementing on/off for basic ones.

Reply:
> But this is not at all an elegant resolution of the issue - I personally would vote for not adding any default connectors instead of adding some of them or implementing on/off for basic ones.

Fully agree